sdk/core/azure-core/azure/core/_utils.py

```diff
@@ -53,6 +53,17 @@ def _convert_to_isoformat(date_time):
         sign, offset = date_time[-6], date_time[-5:]
         delta = int(sign + offset[:1]) * 60 + int(sign + offset[-2:])
 
+    check_decimal = timestamp.split('.')
+    if len(check_decimal) > 1:
+        decimal_str = ""
+        for digit in check_decimal[1]:
+            if digit.isdigit():
+                decimal_str += digit
+            else:
+                break
+        if len(decimal_str) > 6:
+            timestamp = timestamp.replace(decimal_str, decimal_str[0:6])
```

Comment:
I think we have this discussion. If the precision is more than 6, should we just drop the rest?

Reply:
yes - that's what we have decided :)

Reply:
Do we want to log a warning message for precision loss?

Reply:
i don't think it's necessary -  this is an event, there is no scenario where you would re-create a PUT request for ARM and expect precise value.

Reply:
and is it possible that timestamp.replace(decimal_str, decimal_str[0:6]) applies to some other places other than the last segment?

Reply:
nope - it should not